pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -73,6 +77,64 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 	}
 }
 
+// Some versions of enmasse do not handle the address watches correctly
+// this method checks if the watches are handled correctly
+func canWatchEnmasse(cfg *rest.Config) bool {
+	if hasCheckedEnmasse {
+		return canWatchEnmasseVal
+	}
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	log.Info("Checking if can watch enmasse resources")
+	if err == nil {
+		dynamnicAddressResource := dynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "enmasse.io",
+			Version:  "v1beta1",
+			Resource: "addresses",
+		})
```

Comment:
It looks like the new version of enmasse (1.4.0) uses the same GVK for these resources as the previous version (1.3.1), is that correct? If so, is the difference that they're now instances (CRs) of a CRD (and therefore handled by the k8s api server) rather than pre-CRD resources? If so, I wonder if doing a `client.Get` on the CRD would be an alternative approach here?

Reply:
not really.  The dynamic client is the only way to get access to the event channel and it is the immediate useless close of the event channel that signals the error has happened.